packages/material-ui/src/Badge/Badge.js

```diff
@@ -74,19 +84,34 @@ function Badge(props) {
     className,
     color,
     component: ComponentProp,
-    invisible,
+    invisible: invisibleProp,
+    showZero,
+    max,
+    variant,
     ...other
   } = props;
 
+  let invisible = invisibleProp;
+
+  if (invisibleProp == null && Number(badgeContent) === 0 && !showZero) {
+    invisible = true;
+  }
+
   const badgeClassName = classNames(classes.badge, {
     [classes[`color${capitalize(color)}`]]: color !== 'default',
     [classes.invisible]: invisible,
+    [classes.dot]: variant === 'dot',
   });
+  let displayValue = '';
+
+  if (variant !== 'dot') {
+    displayValue = badgeContent > max ? `${max}+` : badgeContent;
```

Comment:
So now `badgeContent` is not a node as specified by the TypeScript/PropTypes definitions, but must be a number?

Reply:
@kLabz Do you have a specific issue using a node? `'hello' > 99 = false`, `({}) > 99 = false`

Reply:
Not really, but this feels really weird. If other node types are still considered supported, it's fine for me :+1:

(Context: I am writing type definitions for a statically typed language so if the type changed I would have updated to the equivalent of `number`)

Reply:
It's true that we don't have any test with a react node. I think that we should add one.